tests/Regression.hs

```diff
@@ -12,8 +19,25 @@ regressionTests :: TestTree
 regressionTests =
     Test.Tasty.testGroup "regression tests"
         [ issue96
+          , unnamedFields
         ]
 
+data Foo = Foo Integer Bool | Bar Bool Bool Bool | Baz Integer Integer
+    deriving (Show, Dhall.Generic, Dhall.Interpret)
+
+unnamedFields :: TestTree
+unnamedFields = Test.Tasty.HUnit.testCase "Unnamed Fields" (do
+    let ty = Dhall.auto @Foo
+    Test.HUnit.assertEqual "Good type" (Dhall.expected ty) (Dhall.Core.Union (
```

Comment:
I believe `Test.Tasty.Hunit` also provides `assertEqual`, so you don't need to pull in the additional `HUnit` dependency

(To be honest, I'm not even sure what's the difference between the two libraries.  I didn't originally author the test suite)